src/components/common/countdown-button/countdown-button.tsx

```diff
@@ -0,0 +1,40 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import type { ButtonProps } from 'react-bootstrap'
+import { Button } from 'react-bootstrap'
+
+export interface CountdownButtonProps extends ButtonProps {
+  countdownSeconds: number
+}
+
+/**
+ * Button that starts a countdown on render and is only clickable after the countdown has finished.
+ */
+export const CountdownButton: React.FC<CountdownButtonProps> = ({ countdownSeconds, children, ...props }) => {
```

Comment:
React-Use provides a hook for intervals. See https://github.com/streamich/react-use/blob/master/docs/useInterval.md